File: ExpressAPIReadOnly_Assignemnt1/models/recipe.js
/**
 * File name: recipe.js
 * Student Name: Kushal Ghimire
 * StudentID: 200555588
 * Date: September 29, 2024
 */

const mongoose = require('mongoose');

// Define the schema for the recipe collection
const recipeSchema = new mongoose.Schema({
  recipeName: { type: String, required: true },
  ingredients: { type: [String], required: true },
  cookingTime: { type: Number, required: true },
  difficulty: { type: String, required: true },
  cuisine: { type: String, required: true },
  description: { type: String, required: true },
  photoLink: { type: String, required: true },
  averageRating: { type: Number, required: true },
});

// Create and export the model
const Recipe = mongoose.model('Recipe', recipeSchema);

module.exports = Recipe;

File: Assignment 3 Android/app/src/main/java/com/kushal/recipefinderapp/model/Recipe.java
package com.kushal.recipefinderapp.model;


import com.google.gson.annotations.SerializedName;

public class Recipe {
    @SerializedName("recipeName")
    private String recipeName;

    @SerializedName("cuisine")
    private String cuisine;

    @SerializedName("description")
    private String description;

    @SerializedName("photoLink")
    private String photoLink;

    public String getRecipeName() {
        return recipeName;
    }

    public String getCuisine() {
        return cuisine;
    }

    public String getDescription() {
        return description;
    }

    public String getPhotoLink() {
        return photoLink;
    }
}
